### PR TITLE
[Enhancement] (nereids)implement showCreateViewCommand in nereids

### LINKED ISSUE #42737: [Enhancement] (nereids)implement showCreateViewCommand in nereids
showCreateView command is supported in legacy planner but not in nereids(new doris fe planner). you need implement the following showCreateView command in nereids:
unsupportedshowCreateViewStatement
``` SHOW CREATE VIEW name=multipartIdentifier                                     ```

### PR DESCRIPTION
**Pull request by msridhar78**
Issue Number: close #42737 

implement showCreateViewCommand in nereids

Regression Test results:
2024-11-16 11:34:36.933 INFO [suite-thread-1] (Suite.groovy:387) - Execute sql: 
            CREATE TABLE IF NOT EXISTS table_for_view_test_2 (
                `key_field` INT COMMENT "Key field",
                `value` STRING COMMENT "Value field"
            )
            DISTRIBUTED BY HASH(key_field) BUCKETS 3
            PROPERTIES ("replication_num" = "1");
        
2024-11-16 11:34:36.951 INFO [suite-thread-1] (Suite.groovy:387) - Execute sql: 
            CREATE VIEW IF NOT EXISTS view_show_create_view_2 AS
            SELECT key_field, value FROM table_for_view_test_2
        
2024-11-16 11:34:36.972 INFO [suite-thread-1] (Suite.groovy:717) - start check3321af53-2873-470e-8231-a57b05fd54e8
2024-11-16 11:34:36.977 INFO [suite-thread-1] (Suite.groovy:721) - start profile run3321af53-2873-470e-8231-a57b05fd54e8
2024-11-16 11:34:36.978 INFO [suite-thread-1] (Suite.groovy:387) - Execute sql: --3321af53-2873-470e-8231-a57b05fd54e8
show create view `view_show_create_view_2`;
2024-11-16 11:34:36.983 INFO [suite-thread-1] (HttpCliAction.groovy:106) - url : http://127.0.0.1:8030/rest/v1/query_profile
2024-11-16 11:34:36.983 INFO [suite-thread-1] (HttpCliAction.groovy:107) - body: null
2024-11-16 11:34:36.983 INFO [suite-thread-1] (HttpCliAction.groovy:108) - op: get
2024-11-16 11:34:36.997 INFO [suite-thread-1] (HttpCliAction.groovy:106) - url : http://127.0.0.1:8030/rest/v1/query_profile/62b9319064f74716-ae9ac1d92192244e
2024-11-16 11:34:36.997 INFO [suite-thread-1] (HttpCliAction.groovy:107) - body: null
2024-11-16 11:34:36.997 INFO [suite-thread-1] (HttpCliAction.groovy:108) - op: get
2024-11-16 11:34:37.069 INFO [suite-thread-1] (Suite.groovy:726) - start profile check3321af53-2873-470e-8231-a57b05fd54e8
2024-11-16 11:34:37.071 INFO [suite-thread-1] (Suite.groovy:727) - Summary:
      -  Profile  ID:  62b9319064f74716-ae9ac1d92192244e
      -  Task  Type:  QUERY
      -  Start  Time:  2024-11-16  11:34:36
      -  End  Time:  2024-11-16  11:34:36
      -  Total:  2ms
      -  Task  State:  EOF
      -  User:  root
      -  Default  Catalog:  internal
      -  Default  Db:  regression_test_nereids_p0_show
      -  Sql  Statement:  --3321af53-2873-470e-8231-a57b05fd54e8
show  create  view  `view_show_create_view_2`;
      -  Distributed  Plan:  N/A
Execution  Summary:
      -  Parse  SQL  Time:  0ms
      -  Nereids  Analysis  Time:  N/A
      -  Nereids  Rewrite  Time:  N/A
      -  Nereids  Optimize  Time:  N/A
      -  Nereids  Translate  Time:  N/A
      -  Nereids  Distribute  Time:  N/A
      -  Workload  Group:  
      -  Analysis  Time:  N/A
      -  Plan  Time:  N/A
          -  JoinReorder  Time:  N/A
          -  CreateSingleNode  Time:  N/A
          -  QueryDistributed  Time:  N/A
          -  Init  Scan  Node  Time:  N/A
          -  Finalize  Scan  Node  Time:  N/A
              -  Get  Splits  Time:  N/A
                  -  Get  Partitions  Time:  N/A
                  -  Get  Partition  Files  Time:  N/A
              -  Create  Scan  Range  Time:  N/A
          -  Get  Partition  Version  Time:  N/A
          -  Get  Partition  Version  Count  (hasData):  N/A
          -  Get  Partition  Version  Count:  N/A
          -  Get  Table  Version  Time:  N/A
          -  Get  Table  Version  Count:  N/A
      -  Schedule  Time:  N/A
          -  Fragment  Assign  Time:  N/A
          -  Fragment  Serialize  Time:  N/A
          -  Fragment  RPC  Phase1  Time:  N/A
          -  Fragment  RPC  Phase2  Time:  N/A
          -  Fragment  Compressed  Size:  0.00  
          -  Fragment  RPC  Count:  0
      -  Schedule  Time  Of  BE:  {}
      -  Wait  and  Fetch  Result  Time:  N/A
      -  Fetch  Result  Time:  0ms
      -  Write  Result  Time:  0ms
      -  Doris  Version:  c9a412b01b
      -  Is  Nereids:  Yes
      -  Is  Cached:  No
      -  Total  Instances  Num:  0
      -  Instances  Num  Per  BE:  
      -  Parallel  Fragment  Exec  Instance  Num:  4
      -  Trace  ID:  
      -  Transaction  Commit  Time:  N/A
      -  System  Message:  N/A
      -  Executed  By  Frontend:  N/A
      -  Nereids  GarbageCollect  Time:  -1ms
      -  Nereids  BeFoldConst  Time:  0ms

Changed  Session  Variables:
VarName                                                                                |  CurrentValue                |  DefaultValue
-----------------------------------------------|---------------------|-------------
character_set_results                                                    |  NULL                                |  utf8mb4          
sql_mode                                                                              |  STRICT_TRANS_TABLES  |                          
variable_version                                                              |  0                                      |  300                  
experimental_enable_nereids_distribute_planner  |  false                              |  true                

2024-11-16 11:34:37.079 INFO [suite-thread-1] (Suite.groovy:1316) - Execute tag: cmd, sql: show create view `view_show_create_view_2`;
2024-11-16 11:34:37.086 INFO [suite-thread-1] (Suite.groovy:387) - Execute sql: DROP VIEW IF EXISTS `view_show_create_view_2`
2024-11-16 11:34:37.094 INFO [suite-thread-1] (Suite.groovy:387) - Execute sql: DROP TABLE IF EXISTS `table_for_view_test_2`
2024-11-16 11:34:37.104 INFO [suite-thread-1] (ScriptContext.groovy:120) - Run test_show_create_view in /root/doris/doris/regression-test/suites/nereids_p0/show/test_show_create_view.groovy succeed
2024-11-16 11:34:58.005 INFO [main] (RegressionTest.groovy:281) - Start to run single scripts
2024-11-16 11:36:25.056 INFO [main] (RegressionTest.groovy:410) - Success suites:
/root/doris/doris/regression-test/suites/nereids_p0/show/test_show_create_view.groovy: group=query,arrow_flight_sql,p0, name=test_show_create_view
2024-11-16 11:36:25.058 INFO [main] (RegressionTest.groovy:492) - All suites success.

### CODE REVIEW

**Comment by msridhar78** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/ShowCreateViewCommand.java [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+    public String getDbName() {
+        return dbName;
+    }
+
+    public String getViewName() {
+        return viewName;
+    }
+
+    @Override
+    public void run(ConnectContext ctx, StmtExecutor executor) throws Exception {
```
Comment (starocean999) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/ShowCreateViewCommand.java:77]: ShowCreateTableStmt's analyze method needs to be re-implemented in nereids. For example, add a private void validate(ConnectContext ctx) throws UserException method to do the job, and call it at the beginning of run method. BTW, the TableNameInfo class has analyze method to do the validation. You'd better keep TableNameInfo as a class member and call its analyze method in the new validate function

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/ShowCreateViewCommand.java [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+        String effectiveCatalog = (catalog == null || catalog.isEmpty()) ? "internal" : catalog;
+
+        // Fetch the catalog, database, and view metadata
+        Env env = Env.getCurrentEnv();
+        DatabaseIf db = env.getCatalogMgr().getCatalogOrAnalysisException(effectiveCatalog)
+                        .getDbOrMetaException(effectiveDbName);
+        TableIf view = db.getTableOrMetaException(viewName);
+
+        if (!(view instanceof View)) {
+            ErrorReport.reportAnalysisException(ErrorCode.ERR_WRONG_OBJECT, effectiveDbName, viewName, "VIEW");
```
Comment (starocean999) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/ShowCreateViewCommand.java:91]: pls use more user friendly error message, take old planner as reference`ErrorReport.reportAnalysisException(ErrorCode.ERR_WRONG_OBJECT, showStmt.getDb(),
                            showStmt.getTable(), "VIEW", "Use 'SHOW CREATE TABLE '" + table.getName());
`

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/ShowCreateViewCommand.java [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+
+        // Lock the view to ensure consistent metadata access
+        view.readLock();
+        try {
+            List<String> createViewStmt = Lists.newArrayList();
+            env.getDdlStmt(null, null, view, createViewStmt, null, null, false, true, false, -1L, false, false);
+
+            if (createViewStmt.isEmpty()) {
+                ShowResultSet resultSet = new ShowResultSet(VIEW_META_DATA, rows);
+                executor.sendResultSet(resultSet);
```
Comment (starocean999) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/ShowCreateViewCommand.java:104]: don't call executor.sendResultSet(resultSet) when holding the readlock.

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/ShowCreateViewCommand.java (hunk 1 above)
Comment (msridhar78) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/ShowCreateViewCommand.java:113]: Thanks for your comment. I have updated the PR, please review

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/ShowCreateViewCommand.java [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+        String effectiveDbName = (tblNameInfo.getDb() == null || tblNameInfo.getDb().isEmpty())
+                        ? ctx.getDatabase() : tblNameInfo.getDb();
+
+        // If catalog is not provided, use the default catalog (e.g., internal)
+        String effectiveCatalog = (tblNameInfo.getCtl() == null || tblNameInfo.getCtl().isEmpty())
+                        ? "internal" : tblNameInfo.getCtl();
+
+        // Fetch the catalog, database, and view metadata
+        Env env = Env.getCurrentEnv();
+        DatabaseIf db = env.getCatalogMgr().getCatalogOrAnalysisException(effectiveCatalog)
```
Comment (starocean999) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/ShowCreateViewCommand.java:108]: keep behavior same as old code:
```suggestion
DatabaseIf db = ctx.getEnv().getCatalogMgr().getCatalogOrAnalysisException(tblNameInfo.getCtl())
                .getDbOrAnalysisException(tblNameInfo.getDb());
```

Comment (msridhar78) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/ShowCreateViewCommand.java]: done - please review

**Comment by starocean999** [comment]
run buildall

**Comment by msridhar78** [comment]
run buildall

**Comment by starocean999** [comment]
run buildall

**Comment by msridhar78** [comment]
run buildall

**Comment by msridhar78** [comment]
run buildall

**Comment by msridhar78** [comment]
run p0

**Comment by msridhar78** [comment]
run buildall

**Comment by msridhar78** [comment]
run buildall

**Review by starocean999 — APPROVED** [review]

**Review by LiBinfeng-01 — APPROVED** [review]

**Pull request merged**